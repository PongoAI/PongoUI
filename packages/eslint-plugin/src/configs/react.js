/** @type {import("eslint").Linter.Config} */
const config = {
  extends: ['airbnb-typescript', 'prettier'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: 'tsconfig.json',
  },
  plugins: ['deprecation', 'import', '@typescript-eslint', 'jest', 'jsx-a11y', 'react', 'react-hooks'],
  settings: {
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      typescript: {
        alwaysTryTypes: true,
      },
    },
    'no-unused-expressions': 'off',
  },
  env: {
    browser: true,
    es6: true,
    node: true,
  },

  ignorePatterns: ['coverage', 'dist', 'dist-storybook', 'lib', 'lib-commonjs', '**/__snapshots__', 'node_modules'],
  rules: {
    'no-unused-vars': 'error',
    'no-use-before-define': 'off',
    '@typescript-eslint/no-use-before-define': ['error'],
    'import/no-extraneous-dependencies': ['error', { devDependencies: true }],
    'import/extensions': [
      'error',
      'ignorePackages',
      {
        js: 'never',
        jsx: 'never',
        ts: 'never',
        tsx: 'never',
      },
    ],
  },
};

module.exports = config;
