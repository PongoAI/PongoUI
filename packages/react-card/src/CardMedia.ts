export * from './components/CardMedia/index';
