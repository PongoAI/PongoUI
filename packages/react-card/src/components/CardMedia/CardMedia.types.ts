import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';

export type CardMediaSlots = {
  /**
   * The root of CardMedia.
   */
  root: Slot<'div'>;

  /**
   * The logo for the media content.
   */
  logo?: Slot<'div'>;
};

export interface CardMediaProps extends ComponentProps<CardMediaSlots> {}

export type CardMediaState = ComponentState<CardMediaSlots>;
