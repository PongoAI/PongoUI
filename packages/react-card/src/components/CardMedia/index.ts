export * from './CardMedia';
export * from './CardMedia.types';
export * from './renderCardMedia';
export * from './useCardMedia';
