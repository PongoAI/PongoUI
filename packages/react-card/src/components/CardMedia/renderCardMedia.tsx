import * as React from 'react';
import { getSlots } from '@fluentui/react-utilities';
import type { CardMediaState, CardMediaSlots } from './CardMedia.types';

export const renderCardMedia = (state: CardMediaState) => {
  const { slots, slotProps } = getSlots<CardMediaSlots>(state);

  return (
    <slots.root {...slotProps.root}>
      {slotProps.root.children}
      {slots.logo && <slots.logo {...slotProps.logo} />}
    </slots.root>
  );
};
