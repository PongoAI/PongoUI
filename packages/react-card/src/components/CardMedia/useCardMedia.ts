import * as React from 'react';
import { getNativeElementProps, resolveShorthand } from '@fluentui/react-utilities';
import type { CardMediaProps, CardMediaState } from './CardMedia.types';

export const useCardMedia = (props: CardMediaProps, ref: React.Ref<HTMLElement>): CardMediaState => {
  const { logo } = props;

  const state: CardMediaState = {
    root: getNativeElementProps('div', {
      ref,
      ...props,
    }),

    components: {
      root: 'div',
      logo: 'div',
    },

    logo: resolveShorthand(logo),
  };

  return state;
};
