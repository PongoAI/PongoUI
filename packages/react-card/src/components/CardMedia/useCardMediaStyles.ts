import { makeStyles, mergeClasses, shorthands } from '@griffel/react';
import type { CardMediaState } from './CardMedia.types';

const useStyles = makeStyles({
  root: {
    position: 'relative',
    ...shorthands.margin('0px -12px'),

    '> *': {
      display: 'block',
      width: '100%',
    },
  },

  logo: {
    position: 'absolute',
    bottom: '12px',
    left: '12px',
    width: '32px',
    height: '32px',
  },
});

export const useCardMediaStyles = (state: CardMediaState) => {
  const styles = useStyles();

  state.root.className = mergeClasses(styles.root, state.root.className);

  if (state.logo) {
    state.logo.className = mergeClasses(styles.logo, state.logo.className);
  }

  return state;
};
