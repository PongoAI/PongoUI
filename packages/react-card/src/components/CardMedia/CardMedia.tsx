import * as React from 'react';
import { useCardMedia } from './useCardMedia';
import { renderCardMedia } from './renderCardMedia';
import { useCardMediaStyles } from './useCardMediaStyles';
import type { CardMediaProps } from './CardMedia.types';
import type { ForwardRefComponent } from '@fluentui/react-utilities';

/**
 * The CardMedia component adds media (image or video) related content to the Card.
 */
export const CardMedia: ForwardRefComponent<CardMediaProps> = React.forwardRef((props, ref) => {
  const state = useCardMedia(props, ref);

  useCardMediaStyles(state);

  return renderCardMedia(state);
});
CardMedia.displayName = 'CardMedia';
