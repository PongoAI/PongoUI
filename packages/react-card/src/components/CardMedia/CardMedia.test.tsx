import * as React from 'react';
import { CardMedia } from './CardMedia';
import { isConformant } from '../../common/isConformant';
import { render } from '@testing-library/react';

describe('CardMedia', () => {
  isConformant({
    Component: CardMedia,
    displayName: 'CardMedia',
  });

  it('renders a basic CardMedia', () => {
    const renderedComponent = render(<CardMedia logo="test" />);

    expect(renderedComponent.baseElement).toMatchSnapshot();
  });
});
