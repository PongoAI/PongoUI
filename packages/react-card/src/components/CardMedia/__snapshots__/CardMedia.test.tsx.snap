// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardMedia renders a basic CardMedia 1`] = `
<body>
  <div>
    <div
      class="___k2tw9a0 f10pi13n fnmsr9s fiot276 f1g46stm f1u10i51 floiv14 f1r7p6p5"
    >
      <div
        class="___d54zkx0 f1euv43f f1gcvs1y f1t6tvco f1szoe96 f1d2rq10"
      >
        test
      </div>
    </div>
  </div>
</body>
`;
